doc/source/rllib/rllib-replay-buffers.rst

```diff
@@ -0,0 +1,145 @@
+the ``replay_buffer_config``.
+
+Basic Usage
+-----------
+
+You will rarely have to define your own replay buffer sub-class, when running an experiment, but rather configure existing buffers.
+The following is `from RLlib's examples section <https://github.com/ray-project/ray/blob/master/rllib/examples/replay_buffer_api.py>`__:  and runs the R2D2 algorithm with `PER <https://arxiv.org/abs/1511.05952>`__ (which by default it doesn't).
+The highlighted lines focus on the PER configuration.
+
+.. dropdown:: **Executable example script**
+    :animate: fade-in-slide-down
+
+    .. literalinclude:: ../../../rllib/examples/replay_buffer_api.py
+        :emphasize-lines: 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77
+        :language: python
+        :start-after: __sphinx_doc_replay_buffer_api_example_script_begin__
+        :end-before: __sphinx_doc_replay_buffer_api_example_script_end__
+
+.. tip:: Because of its prevalence, most Q-learning algorithms support PER. The priority update step that is needed is embedded into their training iteration functions.
+.. warning:: If your custom buffer requires extra interaction, you will have to change the training iteration function, too!
+
+
+Specifying a buffer type works the same way as specifying an exploration type.
+Here are three ways of specifying a type:
+
+.. dropdown:: **Changing a replay buffer configuration**
+    :animate: fade-in-slide-down
+
+    .. literalinclude:: ../../../rllib/examples/documentation/replay_buffer_demo.py
+        :language: python
+        :start-after: __sphinx_doc_replay_buffer_type_specification__begin__
+        :end-before: __sphinx_doc_replay_buffer_type_specification__end__
+
+Apart from specifying a type, other parameters in that config are:
+
+#. Parameters that define how algorithms interact with replay buffers.
+    e.g. ``worker_side_prioritization`` to decide where to compute priorities
+
+#. Constructor arguments to instantiate the replay buffer.
+    e.g. ``capacity`` to limit the buffer's size
+
+#. Call arguments for underlying replay buffer methods.
+    e.g. ``prioritized_replay_beta`` is used by the :py:class:`~ray.rllib.utils.replay_buffers.multi_agent_prioritized_replay_buffer.MultiAgentPrioritizedReplayBuffer` to call the ``sample()`` method of every underlying :py:class:`~ray.rllib.utils.replay_buffers.prioritized_replay_buffer.PrioritizedReplayBuffer`
+
+
+.. tip:: Most of the time, only 1. and 2. are of interest. 3. is an advanced feature that supports use cases where a :py:class:`~ray.rllib.utils.replay_buffers.multi_agent_replay_buffer.MultiAgentReplayBuffer` instantiates underlying buffers that need constructor or default call arguments.
+
+
+ReplayBuffer Base Class
+-----------------------
+
+The base :py:class:`~ray.rllib.utils.replay_buffers.replay_buffer.ReplayBuffer` class only supports storing and replaying
```

Comment:
Mention again here the 2 basic methods that "everyone must know" :)
add and sample